bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Accumulation/AccumulatedSelection.cs

```diff
@@ -0,0 +1,194 @@
+            x => new SelectionShare(x.Value));
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        List<Tuple<ElectionPublicKey, SelectionShare>> guardianShares,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        var lagrangeCoefficients = guardianShares.ComputeLagrangeCoefficients();
+
+        foreach (var (guardian, share) in guardianShares)
+        {
+            Accumulate(share, lagrangeCoefficients[guardian.GuardianId]);
+        }
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        List<Tuple<ElectionPublicKey, SelectionShare>> guardianShares,
+        Dictionary<string, LagrangeCoefficient> lagrangeCoefficients,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        foreach (var (guardian, share) in guardianShares)
+        {
+            Accumulate(share, lagrangeCoefficients[guardian.GuardianId]);
+        }
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        SelectionShare share,
+        LagrangeCoefficient lagrangeCoefficient,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        var copy = new SelectionShare(share);
+        Shares.Add(share.GuardianId, copy);
+        Accumulate(
+            copy.Share,
+            copy.Commitment,
+            lagrangeCoefficient.Coefficient);
+    }
+
+    /// <summary>
+    /// Add a CP proof to the selection. The proof is generated by the guardians through a protocol
+    /// </summary>
+    public void AddProof(ChaumPedersenProof proof)
+    {
+        Proof = new(proof);
+    }
+
+    public bool IsValid()
+    {
+        // TODO: better validation
+        if (Shares.Count == 0)
+        {
+            return false;
+        }
+
+        if (Proof is null)
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    protected override void DisposeManaged()
+    {
+        base.DisposeManaged();
+        Shares.Dispose();
+    }
+
+    protected override void DisposeUnmanaged()
+    {
+        base.DisposeUnmanaged();
+        Value?.Dispose();
+        DescriptionHash?.Dispose();
+        Commitment?.Dispose();
+        Proof?.Dispose();
+    }
+
+    private void Accumulate(
+        ElementModP share,
+        ElGamalCiphertext commitment,
+        ElementModQ lagrangeCoefficient)
+    {
+        // 𝑀𝑏𝑎𝑟 = 𝑀𝑏𝑎𝑟 * (𝑀𝑖 ^ 𝑤𝑖) mod p
+        using var interpolatedshare = BigMath.PowModP(share, lagrangeCoefficient);
+        Value = Value.MultModP(interpolatedshare);
+
+        // a = Πai modp, b = Πbi mod p.
+        Commitment = Commitment.Add(commitment);
```

Comment:
here we accumulate all of the commitments form each share into the aggregate one so that we can make the challenge